dev/com.ibm.ws.security.kerberos.auth/src/com/ibm/ws/security/kerberos/auth/KerberosService.java

```diff
@@ -0,0 +1,79 @@
+/*******************************************************************************
+ * Copyright (c) 2020 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.security.kerberos.auth;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.security.AccessController;
+
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.ConfigurationPolicy;
+
+import com.ibm.websphere.ras.Tr;
+import com.ibm.websphere.ras.TraceComponent;
+import com.ibm.ws.kernel.service.util.SecureAction;
+
+@Component(configurationPolicy = ConfigurationPolicy.REQUIRE,
```

Comment:
I would think the Kerberos service will do more than just get the config and keytab from the kerberos element. So we will know which keytab and config file that we use.
if (kerberos config)
     get keytab and config files
else (get the default file name and location for keytab and config files)

If we have security trace turn on and we can tell which kerberos config/keytab file names that we use, then I am ok with either design. The most common problem we see is which Kerberos config/keytab files that we use.

Reply:
2 questions here:
1) when you say "print out the kerberos config and keytab as an event" do you mean to log the locations being used as an INFO level message? That sounds reasonable to me.

2) Can you elaborate on what you mean by the pseudocode above? I don't think the Liberty code should be doing any parsing of the config/keytab files, or even specifying default values if unspecified. We should just delegate all of the config parsing and default values to whatever the JDK has

Reply:
1) Yes, log the locations being used as an INFO level msg.
2) I am ok with what you have as long as we log the INFO level msg in 1) for the default keytab and config file.